torch/_refs/__init__.py

```diff
@@ -2227,11 +2227,21 @@ def _reduction(
         dims = (dims,)  # type: ignore[assignment]
     dims = utils.reduction_dims(a.shape, dims)
     if not has_identity:
-        valid_shape = a.ndim == 0 or builtins.all(a.shape[i] for i in dims)
+        from torch.fx.experimental.symbolic_shapes import guard_or_true
+
+        valid_shape = a.ndim == 0 or builtins.all(
```

Comment:
here the right thing to do is to create a symbolic big expression and pass it to torch._check(..)
instead of suing guard_or_true. 
see valid_op in this PR for example
https://github.com/pytorch/pytorch/pull/162212